src/__tests__/hooks/medium.useEventOperations.spec.tsx

```diff
@@ -0,0 +1,255 @@
+        createEvent: useCreateEvent(),
+        updateEvent: useUpdateEvent(),
+        deleteEvent: useDeleteEvent(),
+      }),
+      { wrapper }
+    ).result;
+  });
+
+  describe('fetch 초기 상태', () => {
+    it('저장되어있는 초기 이벤트 데이터를 적절하게 불러온다', async () => {
+      // GIVEN: 데이터 불러오기
+      await waitFor(() =>
+        expect(queryEvent.current.fetchEvents.events).toHaveLength(MOCK_EVENTS.length)
+      );
+
+      // THEN: 적확한 초기 이벤트 데이터가 설정되어야 한다.
+      MOCK_EVENTS.forEach((event) =>
+        expect(queryEvent.current.fetchEvents.events).toContainEqual(event)
+      );
+    });
+  });
+
+  describe('update event', () => {
+    it('정의된 이벤트 정보를 기준으로 적절하게 저장이 된다', async () => {
+      // GIVEN: 데이터 불러오기
+      await waitFor(() =>
+        expect(queryEvent.current.fetchEvents.events).toHaveLength(MOCK_EVENTS.length)
+      );
+      // GIVEN: 새롭게 정의된 이벤트 가공
+      const mockUpdateEvent = createEvent({
+        id: '1',
+        title: 'test',
+        startTime: '11:11',
+        endTime: '22:22',
+        description: 'test description',
+        location: 'test location',
+      }) as Event;
+
+      // WHEN: 새롭게 정의된 이벤트를 저장했을 때
+      act(() => queryEvent.current.updateEvent.updateEvent(mockUpdateEvent));
+
+      // THEN: 새롭게 정의된 이벤트를 불러와야 한다.
+      await waitFor(() => {
+        const eventsData = queryEvent.current.fetchEvents.events;
+        expect(eventsData).toContainEqual(mockUpdateEvent);
+      });
+    });
+  });
+
+  describe('create event', () => {
+    it("새로 정의된 'title', 'endTime' 기준으로 적절하게 일정이 업데이트 된다", async () => {
+      // GIVEN: 데이터 불러오기
+      await waitFor(() =>
+        expect(queryEvent.current.fetchEvents.events).toHaveLength(MOCK_EVENTS.length)
+      );
+      // GIVEN: 새로 정의된 데이터 가공
+      const mockCreateEvent = {
+        title: 'test-title',
+        date: '2024-10-15',
+        startTime: '09:00',
+        endTime: '10:00',
+        description: '기본 설명',
+        location: '기본 위치',
+        category: '업무',
+        repeat: { type: 'none', interval: 0, endDate: '' },
+        notificationTime: 1,
+      } as EventForm;
+
+      // WHEN: 새로 정의된 데이터를 업데이트 했을때
+      act(() => queryEvent.current.createEvent.createEvent(mockCreateEvent));
+
+      // THEN: 새로 정의된 데이터를 불러와야 한다.
+      await waitFor(() => {
+        const eventsData = queryEvent.current.fetchEvents.events;
+        const expectedNewEvent = { id: String(MOCK_EVENTS.length + 1), ...mockCreateEvent };
+        expect(eventsData).toContainEqual(expectedNewEvent);
+      });
+    });
+  });
+
+  describe('delete event', () => {
+    it('존재하는 이벤트 삭제 시 에러없이 아이템이 삭제된다.', async () => {
+      // GIVEN: 데이터 불러오기
+      await waitFor(() =>
+        expect(queryEvent.current.fetchEvents.events).toHaveLength(MOCK_EVENTS.length)
+      );
+      MOCK_EVENTS.forEach((event) =>
+        expect(queryEvent.current.fetchEvents.events).toContainEqual(event)
+      );
+
+      // WHEN: 데이터를 삭제 했을 때
+      act(() => queryEvent.current.deleteEvent.deleteEvent('1'));
```

Comment:
비동기 처리가 발생하므로 await를 해주셔야합니다~

```suggestion
      await act(() => queryEvent.current.deleteEvent.deleteEvent('1'));
```